h2o-persist-gcs/src/main/java/water/persist/StringCache.java

```diff
@@ -0,0 +1,33 @@
+package water.persist;
+
+import java.util.ArrayList;
+import java.util.List;
+
+abstract class StringCache {
```

Comment:
This class deserves a test, I think.

Also: i believe this PR brings in dependency on Guava. Guava already provides an implementation of an expiring cache: https://github.com/google/guava/wiki/CachesExplained

Reply:
Using Guava is the best idea @michalkurka ! Concurrency issues are already solved for us, even concurrent attempt to load a value not present in the cache. From LoadingCache's documentation:

```plain
<p>If another call to {@link #get} or {@link #getUnchecked} is currently loading the value for
   * {@code key}, simply waits for that thread to finish and returns its loaded value. Note that
   * multiple threads can concurrently load values for distinct keys.
```